Routers/Routers/Routers/Routers.cs

```diff
@@ -0,0 +1,290 @@
+        if (!IsTheGraphConnected())
+        {
+            throw new DisconnectedGraph();
+        }
+
+        // List of subgraphs, each of which initially contains a vertex from the original graph
+        List<Graph> subgraphs = new();
+
+        // The keys are nodes, the value is the index of the graph with the list in which the node is located
+        Dictionary<int, int> graphInWhichTheNodeIsLocated = new();
+
+        for (int i = 0; i < nodeValue.Count; i++)
+        {
+            subgraphs.Add(new(nodeValue[i]));
+            graphInWhichTheNodeIsLocated.Add(nodeValue[i], i);
+        }
+
+        ///Dictionary of the adjacency of the original graph, but with values sorted in descending order
+        adjacencyDictionary = adjacencyDictionary.OrderBy(x => x.Value, new Comparator()).ToDictionary(x => x.Key, x => x.Value);
+
+        /// We will go by the values in the adjacency dictionary. 
+        /// If two vertices are in different graphs, then we combine 
+        /// them into one graph by combining all existing vertices and adding a new edge. 
+        /// If two vertices already lie in the same graph, 
+        /// then adding an edge with the same vertices will result in a cycle, so in this case we do nothing.
+        /// Eventually there will remain one acyclic graph with the maximum sum of edges
+        while (subgraphs.Count != 1)
+        {
+            // For each pair of vertices from the adjacency dictionary
+            foreach ((int, int) edgeVertices in adjacencyDictionary.Keys)
+            {
+                var (firstNode, secondNode) = edgeVertices;
+                var length = adjacencyDictionary[edgeVertices];
+                if (graphInWhichTheNodeIsLocated[firstNode] == graphInWhichTheNodeIsLocated[secondNode])
+                {
+                    continue;
+                }
+
+                else
+                {
+                    CombineTwoGraphs(firstNode, secondNode, length, subgraphs, graphInWhichTheNodeIsLocated);
+                }
+            }
+        }
+
+        return subgraphs[0];
+    }
+
+    /// <summary>
+    /// Function for combining two graphs
+    /// </summary>
+    /// <param name="firstNode">Value of the first node</param>
+    /// <param name="secondNode">Value of the second node</param>
+    /// <param name="length">Edge length</param>
+    /// <param name="subgraphs">List of graphs</param>
+    private static void CombineTwoGraphs(int firstNode, int secondNode, int length, List<Graph> subgraphs, Dictionary<int, int> dictionary)
+    {
+        Graph firstGraph = subgraphs[dictionary[firstNode]];
+        Graph secondGraph = subgraphs[dictionary[secondNode]];
+
+        for (int i = 0; i < firstGraph.nodeValue.Count; i++)
+        {
+            secondGraph.AddNode(firstGraph.nodeValue[i]);
+        }
+
+        secondGraph.adjacencyDictionary =
+        firstGraph.adjacencyDictionary.Concat(secondGraph.adjacencyDictionary).ToDictionary(x => x.Key, x => x.Value);
```

Comment:
```suggestion
            firstGraph.adjacencyDictionary.Concat(secondGraph.adjacencyDictionary).ToDictionary(x => x.Key, x => x.Value);
```
Перенос  строки всегда выделяется отступом (или даже двумя)